src/AzureBlobStorageUpdateSession/UpdateSession/AzureStorageUpdateSessionDiModule.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using Autofac;
+using Etg.Yams.Azure.Lease;
+using Etg.Yams.Azure.UpdateSession.Retry;
+using Etg.Yams.Update;
+using Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling;
+
+namespace Etg.Yams.Azure.UpdateSession
+{
+    public class AzureStorageUpdateSessionDiModule
+    {
+        private readonly IContainer _container;
+        private const string UpdateSessionRetryStrategyModuleName = "updateSessionRetryStrategy";
+
+        public AzureStorageUpdateSessionDiModule(
+            string clusterId,
+            string instanceId,
+            string updateDomain,
+            string connectionString,
+            TimeSpan updateSessionTtl,
+            int storageExceptionRetryCount = 20,
+            int storageExceptionRetryIntervalInSeconds = 1,
+            int startUpdateSessionRetryCount = 5,
+            int startUpdateSessionRetryIntervalInSeconds = 1) : this(RegisterTypes(
+                clusterId, instanceId, updateDomain, connectionString, updateSessionTtl,
+                storageExceptionRetryCount, storageExceptionRetryIntervalInSeconds,
+                startUpdateSessionRetryCount, startUpdateSessionRetryIntervalInSeconds).Build())
+        {
+        }
+
+        public AzureStorageUpdateSessionDiModule(IContainer container)
+        {
+            _container = container;
+        }
+
+        public static ContainerBuilder RegisterTypes(string clusterId,
+            string instanceId,
+            string updateDomain,
+            string connectionString,
+            TimeSpan updateSessionTtl,
+            int storageExceptionRetryCount = 20,
+            int storageExceptionRetryIntervalInSeconds = 1,
+            int startUpdateSessionRetryCount = 5,
+            int startUpdateSessionRetryIntervalInSeconds = 1)
+        {
+            var containerBuilder = new ContainerBuilder();
+            containerBuilder.RegisterType<BlobLeaseFactory>().As<IBlobLeaseFactory>().SingleInstance();
```

Comment:
Do you still need the BlobLeaseFactory registered (or at all)?

Reply:
Nope, I'll remove it, thanks